src/librustc_mir/interpret/place.rs

```diff
@@ -663,6 +663,23 @@ where
         Ok(())
     }
 
+    /// Write an `Immediate` to memory.
+    #[inline(always)]
+    pub fn write_immediate_to_mplace(
+        &mut self,
+        src: Immediate<M::PointerTag>,
+        dest: MPlaceTy<'tcx, M::PointerTag>,
+    ) -> EvalResult<'tcx> {
+        self.write_immediate_to_mplace_no_validate(src, dest)?;
+
+        if M::enforce_validity(self) {
```

Comment:
this is `false` for the const eval machine, so the check won't happen. Since miri doesn't use `write_immediate_to_mplace`, this is dead code essentially. We're validating in https://github.com/rust-lang/rust/blob/dec0d413446fe3713233584f25490faf65eb15cd/src/librustc_mir/transform/const_prop.rs#L542, so imo it's fine to just call `write_immediate_to_mplace_no_validate` from `const_prop` directly. @RalfJung 	what do you think?

Reply:
Then we'd have to make that function public though.

Reply:
the way it is currently written it is public (just a wrapper), but gives the false impression of actually doing validity checking.

Reply:
I don't think it gives a false impression. It doesn't say `write_immediate_to_place_validate`. It follows the machine's rules for this which seems reasonable for a public interface.

This wa we get validation should we ever decide to turn it on for the CTFE machine.

Reply:
makes sense. Let's keep it as it is.